### PR TITLE
[feat] refactoring of stable mem ops to avoid extra function call  and save code

### PR DESCRIPTION
**Pull request by crusso**
A final refactor of the load/store stable memory operations that avoids a function call to read/write operations (and saves some code)

### CODE REVIEW

File: src/codegen/compile.ml [hunk 1]
```diff
@@ -3805,63 +3805,51 @@ module StableMem = struct
...
             IC.system_call env "ic0" "stable_read" ^^
             get_temp_ptr ^^ load))
     | _ -> assert false
 
-  let write env name typ bytes store =
+  let write env guarded name typ bytes store =
     match E.mode env with
     | Flags.ICMode | Flags.RefMode ->
-      Func.share_code2 env ("__stablemem_write_" ^ name)
+      Func.share_code2 env ("__stablemem_" ^ (if guarded then "guarded_" else "") ^ "write_" ^ name)
```
Comment (ggreif) [inline src/codegen/compile.ml:3836]: I think using `sprintf` would be clearer here.

**Review by ggreif — APPROVED** [review]
Looks good, my comments are low priority!

File: src/codegen/compile.ml [hunk 2]
```diff
@@ -3805,63 +3805,51 @@ module StableMem = struct
           E.else_trap_with env "StableMemory range out of bounds")
     | _ -> assert false
 
-  let read env name typ bytes load =
+  let add_guard env guarded get_offset bytes =
+    if guarded then
+     (if bytes = 1l then
+        get_offset ^^
```
[the remark this answers is not in the document]
Comment (crusso) [reply src/codegen/compile.ml:3811]: Done

File: src/codegen/compile.ml (hunk 1 above)
Comment (crusso) [reply src/codegen/compile.ml:3836]: Done.

**Comment by crusso** [comment]
Merged in claudio/stable-mem

**Pull request closed without merging**